### PR TITLE
(PCP-188) Acceptance test of pxp-module-puppet run

### PR DESCRIPTION
**Pull request by james-stocks**
2nd PR for this test case; contains squashed commits of fixes made in response to first PR  https://github.com/puppetlabs/pxp-agent/pull/295

### CODE REVIEW

File: acceptance/tests/pxp-module-puppet/run_puppet.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+test_name 'C95972 - pxp-module-puppet run' do
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each_with_index do |agent, i|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      cert_dir = configure_std_certs_on_host(agent)
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_test_certs(master, agent, i + 1, cert_dir).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      expect_file_on_host_to_contain(agent, logfile(agent), PXP_AGENT_LOG_ENTRY_ASSOCIATION_SUCCESS, seconds_to_wait = 60)
```
Comment (james-stocks) [inline acceptance/tests/pxp-module-puppet/run_puppet.rb]: @er0ck  - I named the parameter as it's passed. Declaring a variable at the top of the test and referring to it here seemed excessive.

File: acceptance/lib/pxp-agent/test_helper.rb [hunk 2]
```diff
@@ -50,3 +51,102 @@ def kill_pcp_broker(host)
...
+# @param broker the hostname or beaker host object of the machine running PCP broker
+# @param targets array of PCP identities to send request to
+#                e.g. ["pcp://client01.example.com/agent","pcp://client02.example.com/agent"]
+# @param pxp_module which PXP module to call, default pxp-module-puppet
+# @param action which action in the PXP module to call, default run
+# @param params params to send to the module. e.g for pxp-module-puppet:
+#               {:env => [], :flags => ['--noop', '--onetime', '--no-daemonize']}
+# @return hash of responses, with target identities as keys, and value being the response message for that identity as a hash
+# @raise String indicating something went wrong, test case can use to fail
+def rpc_blocking_request(broker, targets,
```
Comment (james-stocks) [inline acceptance/lib/pxp-agent/test_helper.rb]: @er0ck - regarding your comment about whether pcp-client contains methods for what's in this method - no it doesn't, but I've raised PCP-248 to suggest a ruby-pxp-client gem

**Comment by james-stocks** [comment]
@johnduarte @er0ck refreshed the PR

File: acceptance/tests/pxp-module-puppet/run_puppet.rb (hunk 1 above)
Comment (er0ck) [inline acceptance/tests/pxp-module-puppet/run_puppet.rb]: :+1:

File: acceptance/lib/pxp-agent/test_helper.rb (hunk 2 above)
Comment (er0ck) [inline acceptance/lib/pxp-agent/test_helper.rb]: :+1:

**Comment by johnduarte** [comment]
:+1: 
LGTM. Passes on Deb 7.

**Pull request merged**